pkg/grpc/ratelimit/ratelimit_impl_test.go

```diff
@@ -0,0 +1,81 @@
+package ratelimit
+
+import (
+	"math"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"golang.org/x/time/rate"
+)
+
+func TestNewRateLimiterUnlimited(t *testing.T) {
+	rl := NewRateLimiter(-1)
+	assert.Equal(t, rate.Inf, rl.tokenBucketLimiter.Limit())
+
+	rl = NewRateLimiter(0)
+	assert.Equal(t, rate.Inf, rl.tokenBucketLimiter.Limit())
+}
+
+func TestNewRateLimiterLimitedValue(t *testing.T) {
+	expectedRatePerSec := 100
+	rl := NewRateLimiter(expectedRatePerSec)
+
+	assert.Equal(t, rate.Limit(expectedRatePerSec), rl.tokenBucketLimiter.Limit())
+	assert.Equal(t, expectedRatePerSec, rl.tokenBucketLimiter.Burst())
+}
+
+func TestIncreaseLimit(t *testing.T) {
+	noLimitRL := NewRateLimiter(0)
+	assert.Equal(t, rate.Inf, noLimitRL.tokenBucketLimiter.Limit())
+
+	noLimitRL.IncreaseLimit(1)
+	assert.Equal(t, rate.Inf, noLimitRL.tokenBucketLimiter.Limit())
+
+	expectedRatePerSec := 100
+	rl := NewRateLimiter(expectedRatePerSec)
+	assert.Equal(t, rate.Limit(expectedRatePerSec), rl.tokenBucketLimiter.Limit())
+	assert.Equal(t, expectedRatePerSec, rl.tokenBucketLimiter.Burst())
+
+	rl.IncreaseLimit(10)
+	expectedRatePerSec += 10
+	assert.Equal(t, rate.Limit(expectedRatePerSec), rl.tokenBucketLimiter.Limit())
+	assert.Equal(t, expectedRatePerSec, rl.tokenBucketLimiter.Burst())
+}
+
+func TestDecreaseLimit(t *testing.T) {
+	noLimitRL := NewRateLimiter(0)
+	assert.Equal(t, rate.Inf, noLimitRL.tokenBucketLimiter.Limit())
+
+	noLimitRL.DecreaseLimit(1)
+	assert.Equal(t, rate.Inf, noLimitRL.tokenBucketLimiter.Limit())
+
+	expectedRatePerSec := 100
+	rl := NewRateLimiter(expectedRatePerSec)
+	assert.Equal(t, rate.Limit(expectedRatePerSec), rl.tokenBucketLimiter.Limit())
+	assert.Equal(t, expectedRatePerSec, rl.tokenBucketLimiter.Burst())
+
+	rl.DecreaseLimit(10)
+	expectedRatePerSec -= 10
+	assert.Equal(t, rate.Limit(expectedRatePerSec), rl.tokenBucketLimiter.Limit())
+	assert.Equal(t, expectedRatePerSec, rl.tokenBucketLimiter.Burst())
```

Comment:
How about testing decrease with negative input value, with value equal to the current rate, and value greater than current rate (none should impact the token bucket limiter)?